### PR TITLE
accept AbortSignal as an option

### LINKED ISSUE #40: Add the ability to pass options to the emitter (and support AbortController)
For compatible emitters like `node.addEventListener()`.

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#parameters

### PR DESCRIPTION
**Pull request by MoLow**
Fixes #40

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,6 +28,10 @@ export function pEventMultiple(emitter, event, options) {
 			throw new TypeError('The `count` option should be at least 0 or more');
 		}
 
+		if (options.signal) {
+			options.signal.throwIfAborted();
+		}
```
Comment (sindresorhus) [inline index.js:33]:
```suggestion
		options.signal?.throwIfAborted();
```

File: test.js [hunk 2]
```diff
@@ -248,6 +248,36 @@ test('filter option returned with `multiArgs`', async t => {
...
+		emitter.emit('🦄', '🌈');
+	})();
+
+	await t.throwsAsync(pEvent(emitter, '🦄', {signal: AbortSignal.timeout(5)}), {
+		message: 'The operation was aborted due to timeout',
+	});
+	t.is(emitter.listenerCount('🦄'), 0);
+});
+
+test('AbortSignal that is already aborted rejects immidiatly', async t => {
```
Comment (sindresorhus) [inline test.js:265]:
```suggestion
test('AbortSignal that is already aborted rejects immediately', async t => {
```

File: index.d.ts [hunk 3]
```diff
@@ -63,6 +63,14 @@ export interface Options<EmittedType extends unknown | unknown[]> {
 	```
 	*/
 	readonly filter?: FilterFunction<EmittedType>;
+
+	/**
+	 * An `AbortSignal` instance used to cancel the operation.
```
Comment (sindresorhus) [inline index.d.ts:68]: Don't use `*` prefix like the other doc comments.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Thanks for working on this.

You need to add the option to the readme too.

**Comment by MoLow** [comment]
thanks @sindresorhus! any chance you can release a new release to npm?

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/p-event/releases/tag/v6.0.0

**Pull request merged**